lib/Tooling/libSwiftScan/SwiftCaching.cpp

```diff
@@ -956,15 +961,38 @@ static llvm::Error replayCompilation(SwiftScanReplayInstance &Instance,
             if (!Proxy)
               return Proxy.takeError();
 
-            if (Kind == file_types::ID::TY_Object && UseCASBackend)
-              ObjFile = OutputPath->second;
-
             if (Kind == file_types::ID::TY_CachedDiagnostics) {
               assert(!DiagnosticsOutput && "more than 1 diagnostics found");
               DiagnosticsOutput = std::move(*Proxy);
+            } else if (Kind == file_types::ID::TY_SymbolGraphFile &&
```

Comment:
Why is this replay code duplicated?

Reply:
The historical reason is the diagnostics handling part of the replay is really different so this is a separate copy.

I am working on the follow up to eliminate the copy as an improvement. Some handling will be really hard to unify but I think moving the main replay logics into the same file and keep only one copy is the way to go.